internal/ledger/compare.go

```diff
@@ -0,0 +1,413 @@
+			if err != nil {
+				return err
+			}
+		}
+
+	case snapshotRecord2 != nil: // Snapshot 1 is missing a record
+		for snapshotRecord2 != nil {
+			// Add missing to output JSON file
+			diffRecord, err := newDiffRecord(namespace2, nil, snapshotRecord2)
+			if err != nil {
+				return err
+			}
+			err = jsonOutputFile.addRecord(*diffRecord)
+			if err != nil {
+				return err
+			}
+			namespace2, snapshotRecord2, err = snapshotReader2.Next()
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return jsonOutputFile.close()
+}
+
+// diffRecord represents a diverging record in json
+type diffRecord struct {
+	Namespace string          `json:"namespace,omitempty"`
+	Key       string          `json:"key,omitempty"`
+	Record1   *snapshotRecord `json:"snapshotrecord1,omitempty"`
+	Record2   *snapshotRecord `json:"snapshotrecord2,omitempty"`
+}
+
+// Creates a new diffRecord
+func newDiffRecord(namespace string, record1 *privacyenabledstate.SnapshotRecord,
+	record2 *privacyenabledstate.SnapshotRecord) (*diffRecord, error) {
+	// Empty snapshot
+	s0 := snapshotRecord{
+		Value:    "",
+		BlockNum: 0,
+		TxNum:    0,
+	}
+
+	var s1, s2 *snapshotRecord = &s0, &s0 // snapshot records
+	var k string                          // key
+	var err error
+
+	// Snapshot2 has a missing record
+	if record1 != nil {
+		k = stringFromBytes(record1.Key)
+		s1, err = newSnapshotRecord(record1)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Snapshot1 has a missing record
+	if record2 != nil {
+		k = stringFromBytes(record2.Key)
+		s2, err = newSnapshotRecord(record2)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &diffRecord{
+		Namespace: namespace,
+		Key:       k,
+		Record1:   s1,
+		Record2:   s2,
+	}, nil
+}
+
+// snapshotRecord represents the data of a snapshot record in json
+type snapshotRecord struct {
+	Value    string `json:"value,omitempty"`
+	BlockNum uint64 `json:"blockNum,omitempty"`
+	TxNum    uint64 `json:"txNum,omitempty"`
+}
+
+// Creates a new SnapshotRecord
+func newSnapshotRecord(record *privacyenabledstate.SnapshotRecord) (*snapshotRecord, error) {
+	blockNum, txNum, err := heightFromBytes(record.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	return &snapshotRecord{
+		Value:    stringFromBytes(record.Value),
+		BlockNum: blockNum,
+		TxNum:    txNum,
+	}, nil
+}
+
+// Obtain the block height and transaction height of a snapshot from its version bytes
+func heightFromBytes(b []byte) (uint64, uint64, error) {
+	blockNum, n1, err := util.DecodeOrderPreservingVarUint64(b)
+	if err != nil {
+		return 0, 0, err
+	}
+	txNum, _, err := util.DecodeOrderPreservingVarUint64(b[n1:])
+	if err != nil {
+		return 0, 0, err
+	}
+
+	return blockNum, txNum, nil
+}
+
+// Obtain string from bytes
+func stringFromBytes(b []byte) (str string) {
+	str = ""
+	if utf8.Valid(b) {
+		for len(b) > 0 {
+			r, size := utf8.DecodeRune(b)
+			str += string(r)
+			b = b[size:]
+		}
+	} else {
+		str = string(b)
+	}
+
+	return str
+}
```

Comment:
I thought that we had removed this in the previous round as the json encoder internally takes care of this... no?